lib/chef-dk/command/verify.rb

```diff
@@ -141,6 +147,35 @@ def components
           # ------------
           failures = []
 
+          # fail the verify if we have more than one version of chef-provisioning or any of its drivers.
```

Comment:
Yes, I think a hard error is vastly preferable to a warning, which might just go unseen. I just want the error text shown to the user to say

> this test can fail if you've installed new versions of chef provisioning via `chef gem install`, note that some applications may use different versions of chef-provisioning-blah in this situation

... or something like that.